reference/plugins.md

```diff
@@ -2,7 +2,168 @@
 layout: page
 title: Plugins
 ---
-We're working on building a way for you to extend Praxis. Please join the
-[development mailing
-list](https://groups.google.com/forum/#!forum/praxis-development) if you'd like
-to join that conversation.
+
+## Overview
+
+Plugins provide a means to cleanly add new functionality to Praxis, either by extending the core classes with additional features, or by registering callbacks to run during request handling.
+
+For example, extensions to the core classes might include:
```

Comment:
(I suggest changing the this "For example" section and bullets to something like:)

Here are a few examples of what functionality plugins could achieve:
* Adding new DSL directives available within `ActionDefinition`. For exmaple, adding a `requires_authentication` directive to  all `ActionDefinitions` to specify that the action requires an authenticated user.
* Enhancing the `Request` object to carry contextual information. For example, expose information about the currently logged-in user through the Request#current_user method that might return a full User object.
* Enforce application-wide logic. For example, inject a before :action filter into every Controller to perform authentication, or install middleware or around filters for select base controller classes.